presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -305,6 +305,20 @@ private void processNewSources()
         SettableFuture<?> newDriverBlockedFuture = SettableFuture.create();
         driverBlockedFuture.set(newDriverBlockedFuture);
         sourceBlockedFuture.addListener(() -> newDriverBlockedFuture.set(null), newDirectExecutorService());
+
+        // it's possible that memory revoking is requested for some operator
+        // before we update driverBlockedFuture above and we don't want to miss that
+        // notification, so we check to see whether that's the case before returning.
+        boolean memoryRevokingRequested = operators.stream()
+                .map(Operator::getOperatorContext)
```

Comment:
Add `.filter(op -> !revokingOperators.containsKey(operator)`.

Otherwise we will be running `processFor` if there are operators for which `startMemoryRevoke` was called. Such operators would return true `OperatorContext::isMemoryRevokingRequested`.